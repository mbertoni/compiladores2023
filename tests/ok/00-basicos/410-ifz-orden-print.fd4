
let (zero:Nat) = 9 - 9
let (uno :Nat) = 9 - 8

let (tc3:Nat) = let (x:Nat) = 4 in x + 4

let (tc6:Nat) = print "pastito " tc3

let (tc7:Nat) = 9 - 8

let (tc8:Nat) = print "verde" tc6

let (tc11:Nat) = ifz 2+3  then print "Zero = " (2+3) 
                            else print "NotZero = " 2

let (tc12:Nat) = ifz 0  then print "True" zero 
                        else print "False" uno
let (tc13:Nat) = ifz 1  then print "True" zero 
                        else print "False" uno

let (tc14:Nat) = ifz print "Condicion " (2+3)
                    then print "True" 2+7 
                    else print "False" 2

let (tc15:Nat) = ifz zero then tc3 else tc7
let (tc16:Nat) = ifz uno then tc3 else tc7

let (tc17:Nat) = print "verde" tc15

