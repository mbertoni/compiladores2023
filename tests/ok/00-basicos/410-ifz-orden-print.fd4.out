pastito 8
verde8
NotZero = 2
True0
False1
Condicion 5
False2
verde8
